src/backend/codegen/CMakeLists.txt

```diff
@@ -162,26 +183,121 @@ target_link_libraries(
 # Integrate with GPDB build system
 # GPDB unit tests use unittest-check instead of test, so we add an alias target
 # that calls the ctests we registered above.
-add_custom_target(unittest-check ${CMAKE_CTEST_COMMAND})
+add_custom_target(check
+    COMMAND ${CMAKE_CTEST_COMMAND})
+add_custom_target(unittest-check 
+    COMMAND ${CMAKE_COMMAND} .
+    COMMAND ${CMAKE_MAKE_PROGRAM} check
+)
+include_directories(${TOP_SRC_DIR}/src/test/unit/cmockery)
```

Comment:
NOTE: use target_include_directories(TARGET ${DIR}) instead of including cmockery headers for other targets here.